src/Mod/TechDraw/App/DrawViewPart.h

```diff
@@ -90,6 +97,10 @@ class TechDrawExport DrawViewPart : public DrawView
     double getBoxX(void) const;
     double getBoxY(void) const;
     virtual QRectF getRect() const;
+    virtual DrawViewSection* getSectionRef() const;                    //is there a ViewSection based on this ViewPart?
+    Base::Vector3d getUDir(void)  {return uDir;}                       //paperspace X

```

Comment:
uDir, VDir, wDir are  member variables and thus you could return a const-reference to avoid the creation of a tmp. copy.. Also out of const-correctness reasons these methods should be marked as const.